relay-server/src/actors/processor.rs

```diff
@@ -1118,7 +1118,22 @@ impl EnvelopeProcessorService {
             ))),
             _ => ItemAction::Keep,
         });
-        state.has_profile = found_profile;
+        state.has_profile = profile_id.is_some();
+
+        if let Some(event) = state.event.value_mut() {
+            if event.ty.value() == Some(&EventType::Transaction) {
+                let contexts = event.contexts.get_or_insert_with(Contexts::new);
+                // If we found a profile, add its ID to the profile context on the transaction.
+                if let Some(profile_id) = profile_id {
+                    contexts.add(ProfileContext {
+                        profile_id: Annotated::new(profile_id),
+                    });
+                // If not, we delete the profile context.
+                } else {
+                    contexts.remove::<ProfileContext>();
+                }
```

Comment:
If a `ProfileContext` already exists, this code will either create a "duplicated" profile context or remove it -- is this desired? I think these two shouldn't happen because the profile id is taken from the item header, but not sure.

Reply:
`add` actually replaces the existing profile context.